app/Utils/DatabaseCleanupUtils.php

```diff
@@ -137,27 +140,14 @@ public static function removeBuild($buildid) : void
         ", array_merge($buildids, $buildids));
 
         // Delete the configure if not shared.
-        $build2configure = DB::select("
-                               SELECT a.configureid
-                               FROM build2configure AS a
-                               LEFT JOIN build2configure AS b ON (
-                                   a.configureid=b.configureid
-                                   AND b.buildid NOT IN $buildid_prepare_array
-                               )
-                               WHERE a.buildid IN $buildid_prepare_array
-                               GROUP BY a.configureid
-                               HAVING count(b.configureid)=0
-                           ", array_merge($buildids, $buildids));
-
-        $configureids = [];
-        foreach ($build2configure as $build2configure_array) {
-            // It is safe to delete this configure because it is only used
-            // by builds that are being deleted.
-            $configureids[] = intval($build2configure_array->configureid);
-        }
-        if (count($configureids) > 0) {
-            $configureids_prepare_array = $db->createPreparedArray(count($configureids));
-            DB::delete("DELETE FROM configure WHERE id IN $configureids_prepare_array", $configureids);
+        foreach ($builds as $build) {
+            $configure = $build->configure()->first();
+            if ($configure !== null) {
+                $configure_buildids = $configure->builds()->pluck('id')->all();
+                if (array_diff($configure_buildids, $buildids) === []) {
+                    $configure->delete();
+                }
+            }
```

Comment:
What about this cleaner equivalent instead?
```suggestion
            if ($configure !== null && $configure->builds()->count() === 1) {
                $configure->delete();
            }
```

Reply:
This suggestion suffers from the same issue noted in https://github.com/Kitware/CDash/pull/2613#discussion_r1868444420.